src/Diagnostics.DataProviders/DataProviders/Base/SupportObserverDataProviderBase.cs

```diff
@@ -118,6 +111,20 @@ protected async Task<string> GetObserverResource(string url, string resourceId =
             return result;
         }
 
+        protected async Task<HttpResponseMessage> SendObserverRequestAsync(HttpRequestMessage request, string resourceId = null)
+        {
+            request.Headers.TryAddWithoutValidation("x-ms-requestid", RequestId);
```

Comment:
nit: just from consistency point of view, the correct header name is "x-ms-request-id"

Reply:
Oops, typo 🙂

Reply:
If you change this here then also make the change in the LocalHostObserver project so we don't miss out on the logging. Right now LocalHostObserver is expecting a header with name `x-ms-requestid`

Reply:
Ah, that's where I must have copied from